src/equation_sampler/__init__.py

```diff
@@ -190,25 +195,35 @@ def sample_equations(
                 continue
             # sample a tree
             tree = EquationTree(
-                tree_structure, feature_space, function_space, operation_space
+                tree_structure, feature_space, function_space, operator_space
             )
             # sample a valid equation
             tree.sample_valid()
 
             if require_simplify:  # simplify equation
                 current_infix = prefix_to_infix(
-                    tree.expr, function_space, operation_space
+                    tree.expr, function_space, operator_space
                 )
                 if verbose:
                     print("_________")
                     print("infix initial", current_infix)
                     print("initial tree", tree.expr)
-                simplified_equation = simplify(current_infix)
+
+                # create a sympy expression from string
+                expr = sympify(current_infix)
+                symbol_names = [str(symbol) for symbol in expr.free_symbols]
+                real_symbols = symbols(" ".join(symbol_names), real=is_real_domain)
+                if not isinstance(real_symbols, list):
+                    real_symbols = [real_symbols]
+                subs_dict = {old: new for old, new in zip(symbol_names, real_symbols)}
+                expr = expr.subs(subs_dict)
+
+                simplified_equation = simplify(expr)
                 simplified_equation = str(simplified_equation)
                 simplified_equation = simplified_equation.replace(" ", "")
                 simplified_equation = simplified_equation.replace("**", "^")
```

Comment:
this is already in a string so compatible with sympy, so I don't think we need a separate to_sympy. maybe we can return this as well?